src/main/java/io/tiledb/libtiledb/NativeLibLoader.java

```diff
@@ -39,6 +45,27 @@
   /** Path (relative to jar) where native libraries are located. */
   private static final String LIB_RESOURCE_DIR = "/lib";
 
+  /** Temporary directory where native libraries will be extracted. */
+  private static Path tempDir;
+
+  static {
+    try {
+      tempDir = Files.createTempDirectory("tileDbNativeLibLoader");
+    } catch (IOException e) {
+      e.printStackTrace(System.err);
+    }
+    Runtime.getRuntime()
+        .addShutdownHook(
+            new Thread(
+                () -> {
+                  try (Stream<Path> walk = Files.walk(tempDir)) {
+                    walk.sorted(Comparator.reverseOrder()).map(Path::toFile).forEach(File::delete);
```

Comment:
I believe we need delete the directory here too.

Reply:
The final element of the walk will be the directory itself.

Reply:
Thanks for clarifying